grammar.js

```diff
@@ -804,25 +805,34 @@ module.exports = grammar({
     // Primitives
     //
 
-    string: $ => choice(
+    // Here we tolerate unescaped newlines in double-quoted and
+    // single-quoted string literals.
+    // This is legal in typescript as jsx/tsx attribute values (as of
+    // 2020), and perhaps will be valid in javascript as well in the
+    // future.
+    //
+    _string: $ => choice(
       seq(
         '"',
         repeat(choice(
-          token.immediate(prec(PREC.STRING, /[^"\\\n]+|\\\r?\n/)),
+          token.immediate(prec(PREC.STRING, /[^"\\\n]+|\\?\r?\n/)),
           $.escape_sequence
         )),
         '"'
       ),
       seq(
         "'",
         repeat(choice(
-          token.immediate(prec(PREC.STRING, /[^'\\\n]+|\\\r?\n/)),
+          token.immediate(prec(PREC.STRING, /[^'\\\n]+|\\?\r?\n/)),
           $.escape_sequence
         )),
         "'"
       )
     ),
 
+    string: $ => $._string,
+    jsx_string: $ => $._string,
```

Comment:
I think there's no need to have two separate `string` and `jsx_string` rules.

Previously, I had been thinking of a different separation - in which, to be conservative, we would create a separate `jsx_string` rule that allowed unescaped newlines, and would only be used inside of JSX expressions. But even if we did that, I would have used `alias` to make sure that the `jsx_string` appeared as a normal `string` in the output. I don't think it's helpful to have a public-facing `jsx_string` rule, since it's just another node type that users need to remember to target for things like syntax highlighting.

But I think it's probably fine to just allow unescaped newlines in *all* strings, as you've done. I would just get rid of the `jsx_string` thing.

Reply:
Thanks. I will do this. (sorry about rushing this PR and then not reacting - it turns out this is a "hackathon week" at r2c and we're all doing different things this week)

Reply:
I just pushed the fix. There's now just one `string` rule that tolerates unescaped newlines.

Reply:
ping :)